fabric/bin/peer.sh

```diff
@@ -136,7 +136,23 @@ incode/go/example02/cmd' "${OTHER_ARGS[@]}" )
 
 handle_chaincode_instantiate() {
     handle_chaincode_call_with_mapped_name "instantiate" "$@"
-    return
+    # Note: we rely on above to analyze args and assign (& translate) CC_NAME for -n and -c init-args CC_MSG and IS_FPC_CC
+
+    # - instantiate chaincode
+    try $RUN ${FABRIC_BIN_DIR}/peer "${ARGS_EXEC[@]}"
+    # Note: the -c init-args which ultimately have to go to the enclave are also passed here: ecc just ignores them, so no point to filter them. Besides we need it if CC is _not_ FPC ..
+    sleep 3 # unfortunately, no --waitForEvent equivalent, so do heuristic sleep ...
+
+    if [ ${IS_FPC_CC} -eq 1 ]; then
+        # - setup internal ecc state, e.g., register chaincode
+        try $RUN ${FABRIC_BIN_DIR}/peer chaincode invoke -o ${ORDERER_ADDR} -C ${CHAN_ID} -n ${CC_NAME} -c '{"Args":["__setup", "'${ERCC_ID}'"]}' --waitForEvent
```

Comment:
You have a point. However, right now we don't have multi-peer and i also strongly feel for the sdk milestone we should hide setup as it will (should) not be there in the future.  Even in multi-peer setting we still can handle that in the wrapper by persisting info on whether setup has happened and check before invoke and query whether it has happened and if not, call it there.    I think this should be done together with multi-peer, though. Have updated issue #98 accordingly

Reply:
Ok, let's go forward with this strategy. Could you please add a TODO in the cli wrapper for invoke and query setup must be called if needed and link to #98 so we don't miss that?

Reply:
oh, well, if i anyway have to update the patch, i've just fixed it (and as a side-effect also fixed a few bugs in the scripts and made them also more versatile, in particular if you use the interactive version with ledger_init.sh ....